atecc608a_se.c

```diff
@@ -185,103 +187,96 @@ static psa_status_t atecc608a_export_public_key(psa_key_slot_number_t key,
     atecc608a_deinit();
     return status;
 }
-static psa_status_t atecc608a_import_public_key(psa_key_slot_number_t key_slot,
-                                                psa_key_lifetime_t lifetime,
-                                                psa_key_type_t type,
-                                                psa_algorithm_t alg,
-                                                psa_key_usage_t usage,
-                                                const uint8_t *p_data,
-                                                size_t data_length)
+
+static psa_status_t atecc608a_import_public_key(
+    psa_drv_se_context_t *drv_context,
+    psa_key_slot_number_t key_slot,
+    const psa_key_attributes_t *attributes,
+    const uint8_t *data,
+    size_t data_length,
+    size_t *bits)
 {
     const uint16_t key_id = key_slot;
     psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    psa_key_type_t type = psa_get_key_type(attributes);
+    psa_algorithm_t alg = psa_get_key_algorithm(attributes);
 
+    (void) drv_context;
     ASSERT_SUCCESS_PSA(is_public_key_slot(key_id));
 
     /* Check if the key has a size of 65 {0x04, X, Y}. */
     if (data_length != PSA_KEY_EXPORT_MAX_SIZE(PSA_KEY_TYPE_ECC_PUBLIC_KEY(
                                                    PSA_ECC_CURVE_SECP256R1),
-                                               256))
-    {
+                                               256)) {
         return PSA_ERROR_INVALID_ARGUMENT;
     }
 
-    if (type != PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_CURVE_SECP256R1))
-    {
+    if (type != PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_CURVE_SECP256R1)) {
         return PSA_ERROR_NOT_SUPPORTED;
     }
 
     /* The driver can only do randomized ECDSA on SHA-256 */
-    if (alg != PSA_ALG_ECDSA(PSA_ALG_SHA_256) && alg != PSA_ALG_ECDSA_ANY)
-    {
+    if (alg != PSA_ALG_ECDSA(PSA_ALG_SHA_256) && alg != PSA_ALG_ECDSA_ANY) {
         return PSA_ERROR_NOT_SUPPORTED;
     }
 
     ASSERT_SUCCESS_PSA(atecc608a_init());
 
-    ASSERT_SUCCESS(atcab_write_pubkey(key_id, pubkey_for_driver((uint8_t *) p_data)));
+    ASSERT_SUCCESS(atcab_write_pubkey(key_id, pubkey_for_driver((uint8_t *) data)));
+
+    if (bits != NULL) {
+        /* The 64-byte key is stored in the SE as 72 bytes, but we return 64 as
```

Comment:
Minor: why are you calling it a “64-byte” key? It's 65 bytes in the representation used by PSA and 72 bytes inside the ATECC SE. The mathematical object a 256-bit key, which amounts to 32 bytes.

Reply:
The public key data consists of 32 bytes of X followed by 32 bytes of Y, which can be described shortly as a "64 byte public key". Am I wrong?

Reply:
I wanted to modify the comment to explain, why 64 * 8 is returned here, and I put in the comment, what was my understanding of what was happening.

Can you please suggest a wording that would explain, why 64 * 8 is returned here?

Reply:
Oh, I'd noticed that `*bits` was set to the wrong value in an earlier commit, then I mistakenly removed my comment because I thought this had been fixed. I cannot explain `64 * 8` because it's wrong. It should be 256. Thanks for catching this!